tests/countries/test_sint_maarten.py

```diff
@@ -0,0 +1,190 @@
+            "2025-04-27",  # No Sunday observance
+        )
+
+    def test_kings_day_shifted_sunday(self):
+        name = "King's Day"
+        self.assertHolidayName(name, "2014-04-26")  # 2014: April 27 is Sunday
+        self.assertHolidayName(name, "2025-04-26")  # 2025: April 27 is Sunday
+        self.assertNoHoliday("2014-04-27")
+        self.assertNoHoliday("2025-04-27")
+
+    def test_kingdom_day(self):
+        name = "Kingdom Day"
+        # Check that Kingdom Day is not observed before 2011
+        self.assertNoHolidayName(name, range(1900, 2011))
+
+        # Check Kingdom Day from 2011 onward
+        for year in range(2011, 2077):
+            expected_date = get_expected_kingdom_day(year)
+            expected_date_str = expected_date.strftime("%Y-%m-%d")
+            self.assertHolidayName(name, expected_date_str)
+            # If the holiday is moved to Dec 16, ensure Dec 15 is not a holiday
+            if expected_date.day == 16:  # Moved to Monday
+                self.assertNoHolidayName(name, f"{year}-12-15")
+
+    def test_carnival_day(self):
+        name = "Carnival Day"
+        # Check that Carnival Day is not observed before 2011
+        self.assertNoHolidayName(name, range(1900, 2011))
+
+        # Check Carnival Day from 2011 onward
+        for year in range(2011, 2077):
+            expected_date = get_expected_carnival_day(year)
+            expected_date_str = expected_date.strftime("%Y-%m-%d")
+            self.assertHolidayName(name, expected_date_str)
+            # Optionally, ensure the holiday doesn't appear on other dates around April 30
+            for offset in [-2, -1, 1, 2]:  # Check nearby dates
+                test_date = expected_date + timedelta(days=offset)
+                test_date_str = test_date.strftime("%Y-%m-%d")
+                if test_date != expected_date:
+                    self.assertNoHolidayName(name, test_date_str)
+
+    def test_christmas(self):
+        self.assertHolidayName("Christmas Day", (f"{year}-12-25" for year in range(2011, 2077)))
+        self.assertHolidayName(
+            "Second day of Christmas", (f"{year}-12-26" for year in range(2011, 2077))
+        )
+
+
+def get_expected_carnival_day(year):
+    """Calculate the expected date for Carnival Day (April 30) with movement rule."""
+    carnival_date = date(year, 4, 30)
+    weekday = carnival_date.weekday()  # Monday=0, Sunday=6
+    if weekday == 5:  # Saturday
+        return carnival_date - timedelta(days=1)  # Previous Friday
+    elif weekday == 6:  # Sunday
+        return carnival_date + timedelta(days=2)  # Next Tuesday
+    return carnival_date
+
+
+def get_expected_kingdom_day(year):
+    """Calculate the expected date for Kingdom Day (December 15) with movement rule."""
+    kingdom_date = date(year, 12, 15)
+    weekday = kingdom_date.weekday()  # Monday=0, Sunday=6
+    if weekday == 6:  # Sunday
+        return kingdom_date + timedelta(days=1)  # Next Monday
+    return kingdom_date
```

Comment:
I'm not really sure that you need this logic in tests instead of using static dates for checking specific years/dates/holiday names.